File: generators/service-objectStorage/templates/node/instrumentation.js
var IBMCloudEnv = require('ibm-cloud-env');
var ObjectStorage = require('bluemix-objectstorage').ObjectStorage;

module.exports = function(app, serviceManager){
	var region = IBMCloudEnv.getString("objectStorage_region");
	switch (region){
		case "dallas": region = ObjectStorage.Region.DALLAS; break;
		case "london": region = ObjectStorage.Region.LONDON; break;
		default: throw "Invalid Object Storage Region: " + region
	}

	var objStorage = new ObjectStorage({
		projectId: IBMCloudEnv.getString("objectStorage_projectId"),
		userId: IBMCloudEnv.getString("objectStorage_userId"),
		password: IBMCloudEnv.getString("objectStorage_password"),
		region: region
	});

	serviceManager.set("object-storage", objStorage);
};

File: generators/service-objectStorage/templates/python/instrumentation.py
from ibmcloudenv import IBMCloudEnv
import swiftclient

authurl = IBMCloudEnv.getString('objectStorage_auth_url')
if not '/v3' in authurl:
    authurl+='/v3'
user = IBMCloudEnv.getString('objectStorage_userId')
key = IBMCloudEnv.getString('objectStorage_password')
os_options = {
    'project_id': IBMCloudEnv.getString('objectStorage_projectId'),
    'user_id': IBMCloudEnv.getString('objectStorage_userId'),
    'region_name': IBMCloudEnv.getString('objectStorage_region')
}

def getService(app):
    objectStorage = swiftclient.Connection(authurl=authurl,user=user,key=key,os_options=os_options, auth_version='3')

    return 'object-storage', objectStorage
